zebrad/tests/config.rs

```diff
@@ -0,0 +1,342 @@
+
+    let config = ZebradConfig::load(None).expect("load config with ZEBRA_METRICS__ENDPOINT_ADDR");
+    assert_eq!(
+        config.metrics.endpoint_addr.unwrap().to_string(),
+        "0.0.0.0:9999"
+    );
+}
+
+#[test]
+fn zebra_tracing_log_file_env() {
+    let env = EnvGuard::new();
+
+    env.set_var("ZEBRA_TRACING__LOG_FILE", "/test/zebra.log");
+
+    let config = ZebradConfig::load(None).expect("load config with ZEBRA_TRACING__LOG_FILE");
+    assert_eq!(
+        config.tracing.log_file.as_ref().unwrap(),
+        &PathBuf::from("/test/zebra.log")
+    );
+}
+
+#[test]
+fn zebra_mining_miner_address_from_toml() {
+    let _env = EnvGuard::new();
+
+    let miner_address = "u1cymdny2u2vllkx7t5jnelp0kde0dgnwu0jzmggzguxvxj6fe7gpuqehywejndlrjwgk9snr6g69azs8jfet78s9zy60uepx6tltk7ee57jlax49dezkhkgvjy2puuue6dvaevt53nah7t2cc2k4p0h0jxmlu9sx58m2xdm5f9sy2n89jdf8llflvtml2ll43e334avu2fwytuna404a";
+    let toml_string = format!(
+        "[network]\nnetwork = \"Testnet\"\n\n[mining]\nminer_address = \"{}\"",
+        miner_address
+    );
+
+    let mut file = Builder::new()
+        .suffix(".toml")
+        .tempfile()
+        .expect("create temp file");
+    file.write_all(toml_string.as_bytes())
+        .expect("write temp file");
+
+    let config = ZebradConfig::load(Some(file.path().to_path_buf()))
+        .expect("load config with miner_address");
+
+    assert_eq!(
+        config.mining.miner_address.as_ref().unwrap().to_string(),
+        miner_address
+    );
+}
+
+// --- Sensitive env deny-list behaviour ---
+
+#[test]
+fn config_env_unknown_non_sensitive_key_errors() {
+    let env = EnvGuard::new();
+
+    // Unknown field without sensitive suffix should cause an error
+    env.set_var("ZEBRA_MINING__FOO", "bar");
+
+    let result = ZebradConfig::load(None);
+    assert!(
+        result.is_err(),
+        "Unknown non-sensitive env key should error (deny_unknown_fields)"
+    );
+}
+
+#[test]
+fn config_env_unknown_sensitive_key_errors() {
+    let env = EnvGuard::new();
+
+    // Unknown field with sensitive suffix should cause an error
+    env.set_var("ZEBRA_MINING__TOKEN", "secret-token");
+
+    let result = ZebradConfig::load(None);
+    assert!(result.is_err(), "Sensitive env key should cause an error");
+    let msg = result.unwrap_err().to_string();
+    assert!(msg.contains("sensitive key"), "error message: {}", msg);
+}
+
+#[test]
+fn config_env_elasticsearch_password_errors() {
+    let env = EnvGuard::new();
+
+    // This key may or may not exist depending on features. It should be filtered regardless.
+    env.set_var("ZEBRA_STATE__ELASTICSEARCH_PASSWORD", "topsecret");
+
+    let result = ZebradConfig::load(None);
+    assert!(result.is_err(), "Sensitive env key should cause an error");
+
+    #[cfg(feature = "elasticsearch")]
+    {
+        let msg = result.unwrap_err().to_string();
+        assert!(msg.contains("sensitive key"), "error message: {}", msg);
+    }
```

Comment:
This should work regardless of if the feature is selected.

```suggestion
    let msg = result.unwrap_err().to_string();
    assert!(msg.contains("sensitive key"), "error message: {}", msg);
```